src/tiled/variantpropertymanager.cpp

```diff
@@ -135,12 +141,59 @@ int VariantPropertyManager::alignmentTypeId()
     return qMetaTypeId<AlignmentPropertyType>();
 }
 
+QString VariantPropertyManager::objectRefLabel(const MapObject *object) const {
+    QString label = tr("%1: ").arg(QString::number(object->id()));
+    if (!object->name().isEmpty()) {
+        label.append(object->name());
+        if (!object->type().isEmpty())
+            label.append(tr(" (%1)").arg(object->type()));
+    } else if (!object->type().isEmpty())
+        label.append(tr("(%1)").arg(object->type()));
+    else
+        label.append(tr("Unnamed object"));
+
+    return label;
+}
+
 QString VariantPropertyManager::valueText(const QtProperty *property) const
 {
     if (mValues.contains(property)) {
         QVariant value = mValues[property].value;
         int typeId = propertyType(property);
 
+        if (typeId == objectRefTypeId()) {
+            ObjectRef ref = value.value<ObjectRef>();
+
+            if (ref.id == 0)
+                return tr("Unset");
+
+            Document *document = DocumentManager::instance()->currentDocument();
+            if (auto mapDocument = qobject_cast<MapDocument*>(document)) {
```

Comment:
I had commented on this at https://github.com/bjorn/tiled/pull/1408/files#r91930731, but I think you get lucky this time because the Tile Collision Editor has in the meantime been integrated into the tileset editor, as opposed to being a dialog used in the map editor. :-)

I do still wonder if it wouldn't be better to add a `Map*` and `Tile*` to `ObjectRef`.